unittests/test_loader.py

```diff
@@ -75,41 +75,43 @@ def test_load_bad_init(self):
         assert 0 == len(tests)
 
     def test_extended_test(self):
-        with pytest.warns(ReframeDeprecationWarning) as record:
-            @rfm.simple_test
-            class TestSimple(rfm.RegressionTest):
-                # The test should not raise a deprecation warning even though
-                # it overrides __init__
-                def __init__(self):
-                    pass
-
+        with pytest.warns(ReframeDeprecationWarning):
             @rfm.simple_test
             class TestDeprecated(rfm.RegressionTest):
-                # Should raise a warning
                 def setup(self, partition, environ, **job_opts):
                     super().setup(system, environ, **job_opts)
 
+        with pytest.warns(ReframeDeprecationWarning):
             @rfm.simple_test
             class TestDeprecatedRunOnly(rfm.RunOnlyRegressionTest):
                 # Should raise a warning
                 def setup(self, partition, environ, **job_opts):
                     super().setup(system, environ, **job_opts)
 
+        with pytest.warns(ReframeDeprecationWarning):
             @rfm.simple_test
             class TestDeprecatedCompileOnly(rfm.CompileOnlyRegressionTest):
                 # Should raise a warning
                 def setup(self, partition, environ, **job_opts):
                     super().setup(system, environ, **job_opts)
 
+        with pytest.warns(ReframeDeprecationWarning):
             @rfm.simple_test
             class TestDeprecatedCompileOnlyDerived(TestDeprecatedCompileOnly):
-                # Should not raise a warning because the setup of the parent
-                # was not set as final
+                # Should raise a warning
                 def setup(self, partition, environ, **job_opts):
                     super().setup(system, environ, **job_opts)
 
+        with pytest.warns(None) as warnings:
```

Comment:
I think that this is not needed. The test would fail in any case.

Reply:
The test doesn't fail without it, the warnings only appear in the summary like this when I remove the special (...11 passed, 2 warnings in 0.16s...) I think it will look as if it is successful in jenkins, even if it wasn't working correctly

Reply:
Ok, that's fine then. Leave it as is.